File: spring-s3/src/main/java/com/example/springs3/entity/Feed.java
package com.example.springs3.entity;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class Feed {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "feed_id")
    private Long id;

    private String title;

    private String content;

    private String contentImgUrl;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;
    @Builder
    public Feed(String title, String content, String contentImgUrl, User user) {
        this.title = title;
        this.content = content;
        this.contentImgUrl = contentImgUrl;
        this.user = user;
    }
}

File: spring-kotlin/src/main/kotlin/com/example/springkotlin/domain/feed/Feed.kt
package com.example.springkotlin.domain.feed

import com.example.springkotlin.domain.user.User
import jakarta.persistence.*
import java.util.*

@Entity
class Feed(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "feed_id")
    val id: String?=null,

    @Column(nullable = false, length = 20)
    val title:String,

    @Column(nullable = false, length = 255)
    val content:String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    val user: ser
)